packages/mux-uploader/src/mux-uploader-file-select.ts

```diff
@@ -0,0 +1,100 @@
+import { globalThis, document } from 'shared-polyfills';
+const template = document.createElement('template');
+
+template.innerHTML = `
+<style>
+
+button {
+  cursor: pointer;
+  line-height: 16px;
+  background: var(--button-background-color, #fff);
+  border: var(--button-border, 1px solid #000000);
+  color: #000000;
+  padding: var(--button-padding, 16px 24px);
+  border-radius: var(--button-border-radius, 4px);
+  -webkit-transition: all 0.2s ease;
+  transition: all 0.2s ease;
+  font-family: inherit;
+  font-size: inherit;
+  position: relative;
+}
+
+button:hover {
+  color: var(--button-hover-text, #fff);
+  background: var(--button-hover-background, #404040);
+}
+
+button:active {
+  color: var(--button-active-text, #fff);
+  background: var(--button-active-background, #000000);
+}
+
+:host([upload-in-progress]) button {
+  display: none;
+}
+
+:host([upload-in-progress]) ::slotted(button) {
+  display: none;
+}
+</style>
+
+<slot>
+<button type="button">Upload video</button>
+</slot>
+`;
+
+class MuxUploaderFileSelectElement extends globalThis.HTMLElement {
+  protected _filePickerButton: HTMLElement | null | undefined;
```

Comment:
nitpick(blocking): We've been moving to using native javascript `#private` variables. Change for consistency.